src/services/action/base-analytics.ts

```diff
@@ -25,6 +27,31 @@ export class BaseAnalytics implements Analytics {
       requestedSampleSize: number;
     };
   }) {
+    // TODO: all other schemas
+
+    // members: validate and remove additional properties
+    const ajv = new Ajv({ removeAdditional: 'all' });
+    const memberSchema = {
+      type: 'array',
+      items: {
+        // copy of member's schema
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          id: { type: 'string' },
+          name: { type: 'string' },
+          email: { type: 'string' },
+          extra: {
+            type: 'object',
+            additionalProperties: false,
+            properties: { lang: { type: 'string' } },
+          },
+        },
+      },
+    };
```

Comment:
Should this be moved to `src/schemas` and typed accordingly? (you then get static checks of your schema for free, so e.g. VSCode can tell you whether the schema is valid while you're typing it)